### ISSUE TITLE
bug: Unit tests in headless browsers Fail after beta.4

### ISSUE THREAD

**Issue by Barryrowe** [driftyco/ionic#6417]
**Type**: <span ionic-type>bug</span>

**Ionic Version**: <span ionic-version>2.x</span>

**Platform**: <span ionic-platform>all</span>  

<span ionic-description>Due to a change made to point to the Zone symbol for the original 'requestAnimationFrame'/'webkitRequestAnimationFrame' in utils/dom.ts, unit tests running in a headless environment (PhantomJS) end up with an "undefined is not an object" error, because the zone symbols point to undefined functions.

This essentially makes the polyfill found in the lines just before exporting ```raf``` dead code, as the polyfill defines window.requestAnimationFrame after Zone has already done its thing, and then nothing ever returns the polyfill.

I'll be submitting a PR that checks that the function returned from the line below is checked for undefined first, and if so returns the polyfill RAF.

```
(window[window['Zone']['__symbol__']('requestAnimationFrame')] || window[window['Zone']['__symbol__']('webkitRequestAnimationFrame')])
```
[last updated](https://github.com/driftyco/ionic/commit/066ab712c018d9e49005417481ccd2597a349d1d#diff-4929d26d24b00ca6671e0473a2cf27a8R28)</span>

<span is-issue-template></span>

**Comment by Barryrowe** [comment]
The PR submitted for this seems safe, but I understand if it is not merged. It does seem to highlight dead code in the RAF polyfill at the top of dom.ts, but this issue can be solved if the test runner provides a polyfill for RAF before pulling in the zone.js package(s).

Non intrusive Work-around:

1. npm install raf --save-dev
2.  in spec-bundle.js:
```
//...
var raf = require('raf');
raf.polyfill();

require('zone.js/dist/zone');
//...
```

**Comment by Barryrowe** [comment]
This should be fixed with PR #6418 which was merged.
